diff --git a/src/tare_planner/include/planning_env/planning_env.h b/src/tare_planner/include/planning_env/planning_env.h
index 6b574fc..69c2c45 100644
--- a/src/tare_planner/include/planning_env/planning_env.h
+++ b/src/tare_planner/include/planning_env/planning_env.h
@@ -301,6 +301,7 @@ public:
     return pointcloud_manager_->GetNeighborCellsOrigin();
   }
   void GetVisualizationPointCloud(pcl::PointCloud<pcl::PointXYZI>::Ptr vis_cloud);
+  void PublishPlannerCloud();
   void PublishStackedCloud();
   void PublishUncoveredCloud();
   void PublishUncoveredFrontierCloud();
